### ISSUE TITLE
Define a smaller architecture for the student models (e.g., half the size of each parent).

### ISSUE THREAD

**Issue by csmangum** [Dooders/AgentFarm#593]
For AgentFarm’s current `BaseQNetwork` shape, a clean “half-size student” is:

- keep **same input/output dims** (so actions stay compatible),
- reduce hidden width from `H` to `H/2`,
- keep 2 hidden layers (same depth, smaller capacity),
- optionally lower dropout a bit for small models.

```python
import torch
import torch.nn as nn

class StudentQNetwork(nn.Module):
    def __init__(self, input_dim: int, output_dim: int, parent_hidden_size: int = 64) -> None:
        super().__init__()
        student_hidden = max(16, parent_hidden_size // 2)  # half-size with floor

        self.network = nn.Sequential(
            nn.Linear(input_dim, student_hidden),
            nn.LayerNorm(student_hidden),
            nn.ReLU(),
            nn.Dropout(0.05),
            nn.Linear(student_hidden, student_hidden),
            nn.LayerNorm(student_hidden),
            nn.ReLU(),
            nn.Dropout(0.05),
            nn.Linear(student_hidden, output_dim),
        )

        self._initialize_weights()

    def _initialize_weights(self) -> None:
        for m in self.modules():
            if isinstance(m, nn.Linear):
                nn.init.xavier_uniform_(m.weight)
                nn.init.constant_(m.bias, 0.0)

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        if x.dim() == 1:
            x = x.unsqueeze(0)
            return self.network(x).squeeze(0)
        return self.network(x)
```

If you want an even smaller variant for experiments, use `student_hidden = parent_hidden_size // 4` as an ablation.
